### PR TITLE
Updates ES connector

### PR DESCRIPTION
**Pull request by caiooliveiraeti**
1 - Adds pipeline option #266 
2 - Adds PK like id for insert operation
3 - Adds auto generator id for insert operation with consumed message info 
4 - Change pk separator default
5 - Update elastic4s version in ES5 connector

### CODE REVIEW

File: kafka-connect-elastic5/build.gradle [hunk 1]
```diff
@@ -29,6 +30,8 @@ project(":kafka-connect-elastic5") {
             exclude group: "org.scala-lang", module: "scala-library"
             exclude group: "org.scalactic", module: "scalactic_2.11"
         }
+        compile("org.apache.logging.log4j:log4j-core:$log4jVersion")
```
Comment (stheppi) [inline kafka-connect-elastic5/build.gradle]: Don't we already import slf4j?
@Antwnis @andrewstevenson

Comment (andrewstevenson) [reply kafka-connect-elastic5/build.gradle]: I think so I don't think we need to pull it again. @caiooliveiraeti ?

Comment (caiooliveiraeti) [reply kafka-connect-elastic5/build.gradle]: If remove start this error:
```
Cause: java.lang.ClassCastException: org.apache.logging.log4j.simple.SimpleLoggerContext cannot be cast to org.apache.logging.log4j.core.LoggerContext
at org.apache.log4j.LogManager$PrivateManager.getContext(LogManager.java:215)
at org.apache.log4j.LogManager.getRootLogger(LogManager.java:70)
at org.slf4j.impl.Log4jLoggerFactory.<init>(Log4jLoggerFactory.java:66)
at org.slf4j.impl.StaticLoggerBinder.<init>(StaticLoggerBinder.java:72)
at org.slf4j.impl.StaticLoggerBinder.<clinit>(StaticLoggerBinder.java:45)
at org.slf4j.LoggerFactory.bind(LoggerFactory.java:150)
at org.slf4j.LoggerFactory.performInitialization(LoggerFactory.java:124)
```
In elastic4s 6.X this libs is included in TCP project.

Comment (caiooliveiraeti) [reply kafka-connect-elastic5/build.gradle]: If put only:
```
compile("org.apache.logging.log4j:log4j-1.2-api:$log4jVersion")
```
Works too.

**Pull request merged**